File: code/projects/10/ruby/Ostrich.rb
# Ostrich.rb | Defines the Ostrich class which inherits attributes and methods
#   from the WalkingBird superclass.
#
# By: Duncan Van Keulen
# Date:4/27/2020
####################################################

require './WalkingBird.rb'

class Ostrich < WalkingBird

    def call
        'Snork!'
    end

end

File: code/projects/10/java/Ostrich.java
/* Ostrich.java provides an Ostrich subclass of WalkingBird.
 *
 * By: Duncan Van Keulen
 * Date: 4/27/2020
 ******************************************************/

public class Ostrich extends WalkingBird {


    /* default constructor
    */
    public Ostrich() {
        super();
    }

    /* explicit constructor
    * Receive: name, a String 
    * PostCond: myName == name.
    */
    public Ostrich(String name) {
        super(name);
    }

    /* An Ostrich's Call
    * Return: an Ostrich-call ("Whoo-hoo!").
    */
    public String call() {
        return "Snork!";
    }

} 